File: api/Business/Models/Quiz/Quiz.cs
namespace Business.Models
{
    public class Quiz
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public Question[] Questions { get; set; }

        public Quiz()
        {
            Title = "Default Quiz";
            Questions = [];
        }

        public Quiz(int id, string title, Question[] questions)
        {
            Id = id;
            Title = title;
            Questions = questions;
        }
    }
}

File: frontend/src/business/entities/quiz.ts
type BaseQuiz = {
    id: string;
};

export type Quiz = GeneratingQuiz | LoadedQuiz;

export type GeneratingQuiz = BaseQuiz & {
    state: "generating";
}

export type LoadedQuiz = BaseQuiz & {
    state: "loaded";
    title: string;
    questions: Question[];
}

export type QuizDisplay = {
    id: string;
    title: string;
    questionCount: number;
}

export type Question = {
    id: string;
    text: string;
    answers: Answer[];
}

export type Answer = {
    id: string;
    text: string;
    isCorrect: boolean;
};